icgcget/clients/download_client.py

```diff
@@ -56,24 +85,27 @@ def _run_command(self, args, parser, env=None):
             if process.poll() is not None:
                 break
             if output:
-                parser(output.strip())
+                parser(output)
         return_code = process.poll()
         if return_code == 0 and self.session:
             self.session_update('', self.repo)  # clear any running files if exit cleanly
         return return_code
 
     def session_update(self, file_name, repo):
-        for name, file_object in self.session['object_ids'][repo].iteritems():
-            if file_object['index_filename'] == file_name or file_object['filename'] == file_name or \
+        for name, file_object in self.session['file_data'][repo].iteritems():
+            if file_object['index_filename'] == file_name or file_object['fileName'] == file_name or \
                             file_object['fileUrl'] == file_name:
                 file_object['state'] = 'Running'
-                self.session['object_ids'][repo][name] = file_object
+                self.session['file_data'][repo][name] = file_object
             elif file_object['state'] == 'Running':  # only one file at a time can be downloaded.
                 file_object['state'] = 'Finished'
-                self.session['object_ids'][repo][name] = file_object
+                self.session['file_data'][repo][name] = file_object
         json.dump(self.session, open(self.path, 'w', 0777))
 
     def _run_test_command(self, args, forbidden, not_found, env=None, timeout=2):
+        if not env:
+            env = dict(os.environ)
+        env['PATH'] = '/usr/local/bin:' + env['PATH']
```

Comment:
At the very least, this should not be repeated, but encapsulated. But let's see if this is really needed first.